cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -623,6 +623,123 @@
       (let ((amt_ Reg (lo_reg amt)))
         (shr_i128 (put_in_regs src) amt_)))
 
+;;;; Rules for `sshr` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; `i64` and smaller.
+
+(rule (lower (has_type (fits_in_64 ty) (sshr src amt)))
+      (let ((src_ Reg (extend_to_reg src ty (ExtendKind.Sign)))
+            ;; NB: Only the low bits of `amt` matter since we logically mask the
+            ;; shift amount to the value's bit width.
+            (amt_ Reg (lo_reg amt)))
+        (value_reg (sar ty src_ (Imm8Reg.Reg amt_)))))
+
+(rule (lower (has_type (fits_in_64 ty) (sshr src (imm8_from_value amt))))
```

Comment:
I think that `imm8_from_value` isn't the right thing here in the sense that all constants work with this instruction, Cranelift's semantics are just to mask off everything except the low bits. (or at least [I think that's what this previously did](https://github.com/bytecodealliance/wasmtime/blob/1ef0abb12c1978a96b2dd7299390dbe204a138d7/cranelift/codegen/src/isa/x64/lower.rs#L1579-L1581))